sale_global_discount/models/sale_order.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2020 Tecnativa - David Vidal
+# Xtendoo - Manuel Calero
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    global_discount_ids = fields.Many2many(
+        comodel_name='global.discount',
+        string='Sale Global Discounts',
+        domain="[('discount_scope', '=', 'sale'), "
+               "('account_id', '!=', False), '|', "
+               "('company_id', '=', company_id), ('company_id', '=', False)]",
+    )
+    amount_global_discount = fields.Monetary(
+        string='Total Global Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_untaxed_before_global_discounts = fields.Monetary(
+        string='Amount Untaxed Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+    amount_total_before_global_discounts = fields.Monetary(
+        string='Amount Total Before Discounts',
+        compute='_amount_all',
+        currency_field='currency_id',
+        readonly=True,
+    )
+
+    @api.model
+    def get_discounted_global(self, price=0, discounts=None):
+        """Compute discounts successively"""
+        discounts = discounts or []
+        if not discounts:
+            return price
+        discount = discounts.pop(0)
+        price *= 1 - (discount / 100)
+        return self.get_discounted_global(price, discounts)
```

Comment:
using a recursive method here is  inefficient, and it also forces you to make copies of dicounts in other places. 

```suggestion
    def get_discounted_global(self, price=0, discounts=None):
        """Compute discounts successively
        discounts is a list of discount rates, or None
        """
        if discounts is None:
            discounts = []
        for discount in discounts:
            price *= 1 - (discount / 100)
        return price
```